marklogic/src/main/scala/quasar/physical/marklogic/xml/package.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2014–2016 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.physical.marklogic
+
+import quasar.Predef._
+import quasar.Data
+
+import scala.xml._
+
+import scalaz.{Node => _, _}, Scalaz._
+
+package object xml {
+
+  private val attributesKeyName = "_attributes"
+
+  import Data._
+
+  /** Example
+    *
+    * <foo type="baz" id="1">
+    *   <bar>
+    *     <baz>37</baz>
+    *     <bat>one</bat>
+    *     <bat>two</bat>
+    *   </bar>
+    *   <quux>lorem ipsum</quux>
+    * </foo>
+    *
+    * {
+    *   "foo": {
+    *     "_attributes": {
+    *       "type": "baz",
+    *       "id": "1"
+    *     },
+    *     "bar": {
+    *       "baz": "37",
+    *       "bat": ["one", "two"]
+    *     },
+    *     "quux": "lorem ipsum"
+    *   }
+    * }
+    */
+  def toData(elem: Elem): Data = {
+    Obj(ListMap(elem.label -> toDataImpl(elem.child.toList, elem.attributes.some)))
+  }
+
+  private def toDataImpl(nodes: List[Node], m: Option[MetaData]): Data = nodes match {
+    case List(Text(str)) =>
+      m.flatMap(attrToData).cata(
+        m => Obj(ListMap(
+          attributesKeyName -> m,
+          "_text"           -> Str(str)
```

Comment:
How about making the `text` and `attributes` key names configurable, i.e. could be `toData(cfg: ToDataConfig, elem: Elem)` where config is like `ToDataConfig(attributesKeyName: String, textKeyName: String)`.

Then we could have a standard config that just uses `_attributes` and `_text`, respectively.